ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -857,36 +872,38 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = electric $ do
     -- us, as we cannot roll back more than @k@ headers anyway.
     switchTo
       :: HasCallStack
-      => ValidatedChainDiff (Header blk) (LedgerDB' blk)
+      => STM m (ExtLedgerState blk EmptyMK)
+         -- ^ (Volatile) tip of the LedgerDB
+      -> ValidatedChainDiff (Header blk) (Forker' m blk)
          -- ^ Chain and ledger to switch to
       -> StrictTVar m (StrictMaybe (Header blk))
          -- ^ Tentative header
       -> ChainSwitchType
       -> m (Point blk)
-    switchTo vChainDiff varTentativeHeader chainSwitchType = do
+    switchTo tipLegerDB vChainDiff varTentativeHeader chainSwitchType = do
         traceWith addBlockTracer $
             ChangingSelection
           $ castPoint
           $ AF.headPoint
           $ getSuffix
           $ getChainDiff vChainDiff
-        (curChain, newChain, events, prevTentativeHeader) <- atomically $ do
+        (curChain, newChain, events, prevTentativeHeader, newLedger) <- atomically $ do
           curChain  <- readTVar         cdbChain -- Not Query.getCurrentChain!
-          curLedger <- LgrDB.getCurrent cdbLgrDB
+          curLedger <- tipLegerDB
```

Comment:
I'm pretty sure this is an artifact of the code that was already there. I don't see a particular reason (it can't change anywhere else) but I'd rather not change it now and risk finding a super difficult race condition.

Reply:
#1327